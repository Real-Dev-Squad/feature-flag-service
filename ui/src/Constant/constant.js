export const FOOTER = {
    "COPYRIGHT": "Copyright 2022-2023",
    "RDS_WEBSITE": "RDS Website",
    "DISCORD_SERVER":"Discord server"
}