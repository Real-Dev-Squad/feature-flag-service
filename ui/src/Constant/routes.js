import Home from '../Pages/Home/Home';
import Docs from '../Pages/Docs/Docs';
import Sdk from '../Pages/Sdk/Sdk';
import Users from '../Pages/Users/Users';
import Flags from '../Pages/Flags/Flags';
import NotFound from '../Pages/NotFound/NotFound';

export const NAV = [
    {
        name: 'Home',
        path: '/',
        element: <Home />,
        visible: true
    },
    {
        name: "Create Users",
        path:  "/users",
        element:  <Users />,
        visible: true

    },
    {
        name: "Flags",
        path:  "/flags",
        element: <Flags />,
        visible: true
    },
    {
        name: "SDK",
        path:  "/SDK",
        element: <Sdk />
    },
    {
        name: "Docs",
        path:  "/docs",
        element: <Docs />,
        visible: true
    },
    {
        path: "*",
        element: <NotFound />,
        visible: false
    }
]