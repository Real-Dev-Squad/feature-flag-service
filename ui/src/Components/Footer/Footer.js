import { FOOTER } from '../../Constant/constant';

export default function Footer(){
    return(
        <footer className="my-8 w-screen flex justify-center text-xs">
            <span className="ml-2">{FOOTER.COPYRIGHT}</span>
            <a href="#" className="ml-2 hover:underline">{FOOTER.RDS_WEBSITE}</a>
            <a href="#" className="ml-2 hover:underline">{FOOTER.DISCORD_SERVER}</a> 
        </footer>
    )
}