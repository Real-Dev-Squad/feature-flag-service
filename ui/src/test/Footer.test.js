import { render, screen } from '@testing-library/react';
import Footer from './../Components/Footer/Footer';
import { FOOTER} from '../Constant/constant'

describe("Footer component", () => {
    it("Should render footer component",() => {
       render(<Footer />);
    })

    it("Should have RDS Website Link", () => {
        render(<Footer />);
        const RDSLink = screen.getByRole('link',{ name: FOOTER.RDS_WEBSITE });
        expect(RDSLink).toBeInTheDocument();
    })

    it("Should have RDS Discord Link", () => {
        render(<Footer />);
        const RDSLink = screen.getByRole('link',{ name: FOOTER.DISCORD_SERVER });
        expect(RDSLink).toBeInTheDocument();
    })

    it("Should have 2 Link", () => {
        render(<Footer />);
        const RDSLink = screen.getAllByRole('link');
        expect(RDSLink).toHaveLength(2);
    })

    it("Should have copyright text", () => {
        render(<Footer />);
        const copyrightLink = screen.getByText(FOOTER.COPYRIGHT);
        expect(copyrightLink).toBeInTheDocument();
    })

    it("Should match the snapshot", () =>{
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
      })
})