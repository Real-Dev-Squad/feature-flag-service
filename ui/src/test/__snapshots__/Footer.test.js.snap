// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should match the snapshot 1`] = `
<div>
  <footer
    class="my-8 w-screen flex justify-center text-xs"
  >
    <span
      class="ml-2"
    >
      Copyright 2022-2023
    </span>
    <a
      class="ml-2 hover:underline"
      href="#"
    >
      RDS Website
    </a>
    <a
      class="ml-2 hover:underline"
      href="#"
    >
      Discord server
    </a>
  </footer>
</div>
`;
